internal/pkg/views/list.go

```diff
@@ -113,15 +121,44 @@ func (w *ListWidget) itemsToShow() []*expanders.TreeNode {
 	return w.currentPage.FilteredItems
 }
 
-func highlightText(displayText string, highlight string) string {
+func highlightText(displayText string, highlight string, filterFuzzy bool) string {
 	if highlight == "" {
 		return displayText
 	}
-	index := strings.Index(strings.ToLower(displayText), highlight)
-	if index < 0 {
-		return displayText
+	if filterFuzzy {
+		result := ""
+		displayLower := strings.ToLower(displayText)
+		highlightIndex := 0
+		highlightLower := strings.ToLower(highlight)
+		highlightCharLower := rune(highlightLower[highlightIndex])
+
+		for displayIndex, displayCharLower := range displayLower {
+			if displayCharLower == highlightCharLower {
+				// got a match - highlight this character (from the non-lowered display text)
+				result += style.Highlight(string(displayText[displayIndex]))
+				highlightIndex++
+				if highlightIndex < len(highlight) {
+					highlightCharLower = rune(highlightLower[highlightIndex])
+				} else {
+					// we're done, add any remaining characters
+					if displayIndex+1 < len(displayText) {
+						result += displayText[displayIndex+1:]
+					}
+					break
+				}
+			} else {
+				// not a match - add this character (from the non-lowered display text)
+				result += string(displayText[displayIndex])
+			}
```

Comment:
👆 this is fun!

Reply:
I'm open to suggestions for improvements. This got a little gnarly as I wanted case-insensitive matching but preserving the case in the output!

Reply:
I've made a change to the implementation of this that I think simplifies it slightly - let me know what you think :-)